src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs

```diff
@@ -846,70 +1214,118 @@ public class Type { }
+                // 1.cs(7,28): error CS1950: The best overloaded Add method 'ListBase<int>.Add(string)' for the collection initializer has some invalid arguments
+                //         ListBase<int> y = [1, 2];
+                Diagnostic(ErrorCode.ERR_BadArgTypesForCollectionAdd, "1").WithArguments("System.Collections.Generic.ListBase<int>.Add(string)").WithLocation(7, 28),
+                // 1.cs(7,28): error CS1503: Argument 1: cannot convert from 'int' to 'string'
+                //         ListBase<int> y = [1, 2];
+                Diagnostic(ErrorCode.ERR_BadArgType, "1").WithArguments("1", "int", "string").WithLocation(7, 28),
+                // 1.cs(7,31): error CS1950: The best overloaded Add method 'ListBase<int>.Add(string)' for the collection initializer has some invalid arguments
+                //         ListBase<int> y = [1, 2];
+                Diagnostic(ErrorCode.ERR_BadArgTypesForCollectionAdd, "2").WithArguments("System.Collections.Generic.ListBase<int>.Add(string)").WithLocation(7, 31),
+                // 1.cs(7,31): error CS1503: Argument 1: cannot convert from 'int' to 'string'
+                //         ListBase<int> y = [1, 2];
+                Diagnostic(ErrorCode.ERR_BadArgType, "2").WithArguments("1", "int", "string").WithLocation(7, 31));
         }
 
         [Fact]
-        public void DelegateType_01()
+        public void ListInterfaces_01()
         {
-            string source = """
-                delegate void D();
+            string sourceA = """
+                namespace System
+                {
+                    public class Object { }
+                    public abstract class ValueType { }
+                    public class String { }
+                    public class Type { }
+                    public struct Void { }
+                    public struct Boolean { }
+                    public struct Int32 { }
+                }
+                namespace System.Collections
+                {
+                    public interface IEnumerable { }
+                }
+                namespace System.Collections.Generic
+                {
+                    public interface IA { }
+                    public interface IB<T> { }
+                    public interface IC<T> { }
+                    public interface ID<T1, T2> { }
+                    public class List<T> : IEnumerable, IA, IB<T>, IC<object>, ID<T, object>
+                    {
+                        public void Add(T t) { }
+                    }
+                }
+                """;
+            string sourceB = """
+                using System.Collections.Generic;
                 class Program
                 {
                     static void Main()
                     {
-                        D d;
-                        d = [];
-                        d = [1, 2];
+                        List<int> l = [1];
+                        IA a = [2];
```

Comment:
Why isn't this one constructible? And what does that imply about `IEnumerable i = [1, 2, 3];`?

Reply:
The only interfaces that are considered constructible are interfaces implemented by `List<T>` with one type parameter, where the type argument is `T`. `IEnumerable<T>` is constructible, but `IEnumerable` is not.